File: 卡码/代码随想录/JS/二叉树/01-二叉树遍历方式/LeetCode_0111.js
//111. 二叉树的最小深度
import TreeNode from "../TreeNode.js";

const minDepth = function(root) {
  q=new Array();
  let depth=0;
  if(root) q.push(root);
  while(q.length>0){
    let size=q.length;
    depth++;
    while(size-->0){
      const node=q.shift();
      if(node.left) q.push(node.left);
      if(node.right) q.push(node.right);
      if(!node.left&&!node.right) return depth;
    }
  }
  return depth;
};

File: 卡码/代码随想录/Python/二叉树/01-二叉树遍历方式/LeetCode_0111.py
#111. 二叉树的最小深度
from typing import *
from ..TreeNode import *

class Solution:
  def minDepth(self, root: Optional[TreeNode]) -> int:
    depth:int=0
    q:List[TreeNode]=[]
    if root!=None:
      q.append(root)
    while len(q)>0:
      size:int=len(q)
      depth+=1
      for i in range(size):
        node:TreeNode=q[0]
        q=q[1:]
        if node.left:
          q.append(node.left)
        if node.right:
          q.append(node.right)
        if not node.left and not node.right:
          return depth
    return depth
